tests/csm/rest/test_query_deployment.py

```diff
@@ -0,0 +1,112 @@
+        self.log.info("Prerequisite: Deploy cortx cluster")
+        self.log.info("Cleanup: Destroying the cluster ")
+        resp = self.deploy_lc_obj.destroy_setup(self.csm_obj.master, self.csm_obj.worker_list,
+                                                K8S_SCRIPTS_PATH)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Cleanup: Cluster destroyed successfully")
+
+        self.log.info("Cleanup: Setting prerequisite")
+        self.deploy_lc_obj.execute_prereq_cortx(self.csm_obj.master,
+                                                K8S_SCRIPTS_PATH,
+                                                K8S_PRE_DISK)
+
+        for node in self.csm_obj.worker_list:
+            self.deploy_lc_obj.execute_prereq_cortx(node, K8S_SCRIPTS_PATH,
+                                                    K8S_PRE_DISK)
+        self.log.info("Cleanup: Prerequisite set successfully")
+
+        self.log.info("Cleanup: Deploying the Cluster")
+        resp_cls = self.deploy_lc_obj.deploy_cluster(self.csm_obj.master,
+                                                     K8S_SCRIPTS_PATH)
+        assert_utils.assert_true(resp_cls[0], resp_cls[1])
+        self.log.info("Cleanup: Cluster deployment successfully")
+
+        self.log.info("[Start] Sleep %s", self.update_seconds)
+        time.sleep(self.update_seconds)
+        self.log.info("[Start] Sleep %s", self.update_seconds)
+
+        self.log.info("Cleanup: Check cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.csm_obj.master)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Cleanup: Cluster status checked successfully")
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-45672')
+    def test_45675(self):
+        """
+        Verify GET cluster topology with valid storage id
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Open and parse solution yaml")
+        self.log.info("Step 1: Send GET request for fetching system topology"
+                      "without storage set id")
+        resp, result = self.csm_obj.verify_storage_set()
+        assert result, "Verification for get system topology failed."
+        self.log.info("Response : %s", resp)
+        self.log.info("Step 2: Send GET request for fetching system topology"
+                      "with storage set id")
+        storage_sets = resp.json()["topology"]["cluster"]["storage_set"]
+        res = [sub['id'] for sub in storage_sets]
+        print("Printing list of storage sets : " + str(res))
+        for storage_set_id in storage_sets:
+            self.log.info("Sending request for %s ", storage_set_id)
+            resp, result = self.csm_obj.verify_storage_set(storage_set_id)
+            assert result, "Verification for get system topology failed."
```

Comment:
Error message should exactly point what has failed

Reply:
done